docs/DeveloperGuide.md

```diff
@@ -1187,28 +1196,92 @@ The `MathUtil` class contains generic functions for mathematical operations such
 
 ---
 
-## **Unit Tests**
+## **Testing Types**
+
+Types of tests used:
+* Developer Testing
+* Regression Testing
+* Unit Tests
+* Stubs
+* Integration testing
+* Dogfooding
+
+Our organisation made use of different testing types in order to cover multiple grounds. Types of testings also changed 
+with the progress of the product.
+
+### **Developer Testing**
+
+Our organisation opted to use developer testing right from the start with every increment of the product, instead of 
+waiting for the whole product to be ready before conducting more tests.
+
+This type of testing involved the developers trying out the product with the various commands implemented, to 
+discover possible bugs early, and avoid major reworks in future increments and iterations.
+
+### **Regression Testing**
+
+Our organisation continued with regression testing as increments came in, to ensure that previous features were not 
+broken with changes and modifications. This is due to the interconnectedness of certain features implemented, such as 
+commands and the model. We used a breadth-first iterative model to extensive test the wide scope of the project.
+
+Our organisation did this both with automated unit tests and manual testing, with more emphasis on unit tests, due to 
+the time taken to test extensively manually. 
+
+### **Unit Tests**
 
 Our organisation primarily utilises `JUnit5` for conducting unit tests. `JUnit5` is a popular testing framework in the Java programming language that enables developers to write and run unit tests for their code.
 
 Each individual unit of the program is tested to ensure each piece works correctly and stubs are commonly used to test the units in isolation.
 
-### **Testing Models**
+Unit tests were implemented as new features were added in, to ensure that any new changes will have tests to ensure 
+older features are working as intended. Unit tests were also changed and added on as the product progressed, to ensure
+that the features were tested more thoroughly as the team learned new testing types and techniques.
+
+
+### **Stubs**
+
+Stubs were used to test features and classes in isolation, so that any bugs in dependencies will not affect the tests
+that are in place. This is most common in the testing of commands as well as parsers, to ensure that any bugs in one or
+the other will not result in failing tests in the other, allowing us to quickly ascertain bugs without having to find 
```

Comment:
This is not the reason why we use stubs in testing Commands and Parsers. It is because they have a lot of dependencies which make it hard to test the entire object itself. You can add on stubs help to isolate dependencies so as to ... (write yourself)